builtin/rebase--helper.c

```diff
@@ -10,22 +10,36 @@ static GIT_PATH_FUNC(path_squash_onto, "rebase-merge/squash-onto")
 
 static int get_revision_ranges(const char *upstream, const char *onto,
 			       const char **head_hash,
-			       char **revisions)
+			       char **revisions, char **shortrevisions)
 {
+	const char *shortupstream, *shorthead;
 	struct object_id orig_head;
 
 	if (get_oid("HEAD", &orig_head))
 		return error(_("no HEAD?"));
 
 	*head_hash = find_unique_abbrev(&orig_head, GIT_MAX_HEXSZ);
 
+	if (shortrevisions)
+		shorthead = find_unique_abbrev(&orig_head, DEFAULT_ABBREV);
+
 	if (upstream) {
 		struct object_id upstream_oid;
 
 		get_oid(upstream, &upstream_oid);
-		*revisions = xstrfmt("%s...%s", upstream, *head_hash);
-	} else
-		*revisions = xstrfmt("%s...%s", onto, *head_hash);
+
+		if (revisions)
+			*revisions = xstrfmt("%s...%s", upstream, *head_hash);
+		if (shortrevisions) {
+			shortupstream = find_unique_abbrev(&upstream_oid, DEFAULT_ABBREV);
```

Comment:
shortupstream is used only inside the current if clause, so it could be declared here too.